src/ripple/app/ledger/LedgerTiming.h

```diff
@@ -20,42 +20,108 @@
 #ifndef RIPPLE_APP_LEDGER_LEDGERTIMING_H_INCLUDED
 #define RIPPLE_APP_LEDGER_LEDGERTIMING_H_INCLUDED
 
+#include <cstdint>
+#include <utility>
+
 namespace ripple {
 
+/** Determines whether the current ledger should close at this time.
+
+    This function should be called when a ledger is open and there is no close
+    in progress, or when a transaction is received and no close is in progress.
+
+    @param anyTransactions indicates whether any transactions have been received
+    @param previousProposers proposers in the last closing
+    @param proposersClosed proposers who have currently closed this ledger
+    @param proposersValidated proposers who have validated the last closed ledger
+    @param previousMSeconds time, in milliseconds, for the previous ledger to
+                            reach consensus (in milliseconds)
+    @param currentMSeconds time, in milliseconds since the previous ledger closed
+    @param openMSeconds time, in milliseconds, since the previous LCL was computed
+    @param idleInterval the network's desired idle interval
+*/
+bool shouldCloseLedger (
+    bool anyTransactions,
+    int previousProposers,      int proposersClosed,    int proposerersValidated,
```

Comment:
if you had one variable declaration per line, you'd have instantly spotted the spelling mistake `proposerersValidated`.  :-)